src/shared/esp-now-structs.h

```diff
@@ -74,6 +77,12 @@ struct EspNowChallengeFirmwareResponseV1 {
   char md5[32];                 // MD5 hash of firmware. Does not include trailing \0
 };
 
+struct EspNowChallengeConfigResponseV1 {
+  uint8_t id = MESSAGE_ID_CHALLENGE_CONFIG_RESPONSE_V1;
+  uint32_t challenge_challenge; // Challenge from [EspNowChallengeRequestV1].
+  EspNowConfigEnvelope envelope;
```

Comment:
I've made changes here.  Because the EspNowNode needs to fetch the configuration version and length from NVS, without having to fetch the data, I've got a ConfigurationHeader struct with these two pieces of data.  The Node itself fetches both the header and the data.

I've likely committed some cpp/embedded crimes in this area, e.g. the Host code does an allocation off the heap and EspNowHost deletes it.  That was because I couldn't figure out how to have the Host return by value a struct that contains a buffer of arbitrary length.